File: Code/RaspberryPi/opencv/showmovie/showmovie.py
#Import libraries
import cv2
import os

#Counter Variable
count = 0

#img = cv2.imread('/home/pi/robot/Media/Pictures/Robot03.jpg')
#cv2.imshow('image',img)

#cv2.namedWindow('Frames', cv2.WINDOW_AUTOSIZE)
cv2.namedWindow('Frames')

cap = cv2.VideoCapture('/home/pi/robot/Media/Video/RobotVideo01.mp4')
#cap = cv2.VideoCapture(0)

while (cap.isOpened()):
   # Capture frame-by-frame
   ret, frame = cap.read()
   if ret == True:
      print('Read %d frame: ' % count, ret)
      cv2.imshow('Frames',frame)
      count += 1
   else:
      break

   key = cv2.waitKey(3)
   if key == 27:#if ESC is pressed, exit loop
        cv2.destroyAllWindows()
        break

# When everything done, release the capture
cap.release()
print(count)

File: Code/RaspberryPi/opencv/showmovie/showmovie.cpp
//----------------------------------------------------------------------------
//
//  $Workfile: ColorLocator.cpp$
//
//  $Revision: X$
//
//  Project:    Open CV example
//
//                            Copyright (c) 2017
//                               James A Wright
//                            All Rights Reserved
//
//  Modification History:
//  $Log:
//  $
//
//  Notes:
//     This is a tool for tuning a camera
//
//----------------------------------------------------------------------------
#include "opencv2/core.hpp"
#include "opencv2/imgproc.hpp"
#include "opencv2/highgui.hpp"
#include <stdio.h>
#include <iostream>

#define FROM_IMAGE

using namespace cv;
using namespace std;

//----------------------------------------------------------------------------
//  File Locals
//----------------------------------------------------------------------------
int mMorphSize = 7;
vector<vector<Point> > mContours;
vector<Vec4i> mHierarchy;
Rect mRectangle;
Mat mSourceImage;

//--------------------------------------------------------------------
// Purpose:
//     Main entry point
//
// Notes:
//     None.
//--------------------------------------------------------------------
int main(int argc, char** argv)
{
    VideoCapture cap("/home/pi/robot/Media/Video/RobotVideo01.mp4");
//    VideoCapture cap(0); //capture the video from web cam
    // Can't open the camera
    if (!cap.isOpened())  // if not success, exit program
    {
        cout << "Cannot open the web cam" << endl;
        return -1;
    }

    bool bSuccess = cap.read(mSourceImage); // read a new frame from video
   
    // Idle loop
    while (true)
    {
        bool bSuccess = cap.read(mSourceImage); // read a new frame from video

        if (!bSuccess) //if not success, break loop
        {
            cout << "Cannot read a frame from video stream" << endl;
            break;
        }

        imshow("Original", mSourceImage); //show the original image

        //wait for 'esc' key press for 30ms. If 'esc' key is pressed, break loop
        if (waitKey(30) == 27) 
        {
            cout << "esc key is pressed by user" << endl;
            break;
        }
    }

    return 0;
}
